File: Python/array/find_pivot_index.py
def pivotIndex(nums: list) -> int:
    if not nums:
        return -1

    total = 0
    for num in nums:
        total += num
    
    precent = 0
    for i in range(len(nums)):
        if precent == total - nums[i] - precent:
            return i
        precent += nums[i]
    
    return -1

if __name__ == "__main__":
    print(pivotIndex([1, 2, 3]))

File: Java/src/com/dbc/algorithm/Array/FindPivotIndex.java
package com.dbc.algorithm.Array;

public class FindPivotIndex {
    public static int pivotIndex(int[] nums) {
        if (nums.length == 0) {
            return -1;
        }

        int total = 0;
        for (int i = 0; i < nums.length; i++) {
            total += nums[i];
        }

        int present = 0;
        for (int i = 0; i < nums.length; i++) {
            if (present == total - present - nums[i]) {
                return i;
            }
            present += nums[i];
        }

        return -1;
    }
}
